metron-interface/metron-config/e2e/sensor-config/sensor-config.po.ts

```diff
@@ -0,0 +1,243 @@
+    return element(by.cssContainingText(fieldRowClassName, name)).element(by.xpath("..")).element(by.css('.fa-pencil'));
+  }
+
+  getFormData() {
+    let mainPanel = element.all(by.css('.metron-slider-pane-edit')).last();
+    return protractor.promise.all([
+      mainPanel.element(by.css('.form-title')).getText(),
+      mainPanel.element(by.css('input[name="sensorTopic"]')).getAttribute('value'),
+      mainPanel.element(by.css('select[formcontrolname="parserClassName"]')).getAttribute('value'),
+      mainPanel.element(by.css('input[formcontrolname="grokStatement"]')).getAttribute('value'),
+      mainPanel.all(by.css('div[name="fieldSchema"] table tr')).getText(),
+      mainPanel.all(by.css('div[name="threatTriage"] table tr')).getText(),
+      mainPanel.element(by.css('input[formcontrolname="index"]')).getAttribute('value'),
+      mainPanel.element(by.css('metron-config-number-spinner[name="batchSize"] input')).getAttribute('value'),
+      mainPanel.all(by.css('metron-config-advanced-form input')).getAttribute('value')
+    ]).then(args => {
+      return  {
+        title: args[0],
+        parserName: args[1],
+        parserType: args[2],
+        grokStatement: args[3],
+        fieldSchemaSummary: args[4],
+        threatTriageSummary: args[5],
+        indexName: args[6],
+        batchSize: args[7],
+        advancedConfig: args[8]
+      }
+    });
+  }
+
+  getKafkaStatusText() {
+    return element(by.css('input[name="sensorTopic"] + label')).getText();
+  }
+
+  getThreatTrigaeRule() {
+    return element.all(by.css('.threat-triage-rule-row.threat-triage-rule-str')).getText();
+  }
+
+  getThreatTriageSummary() {
+    return element.all(by.css('div[name="threatTriage"] table tr')).getText();
+  }
+
+  getTransformText() {
+    return element.all(by.css('.config.container')).all(by.css('.edit-pane-readonly')).getText();
+  }
+
+  navigateTo(parserName: string) {
+    let url = browser.baseUrl + '/sensors(dialog:sensors-readonly/'+ parserName + ')';
+    return changeURL(url);
+  }
+
+  setAdvancedConfig(key: string, value: string) {
+    return element.all(by.css('.advanced-link')).click().then(() => {
+      element(by.css('input[formcontrolname="newConfigKey"]')).sendKeys(key);
+      element(by.css('input[ng-reflect-name="newConfigValue"]')).sendKeys(value);
+    });
+  }
+
+  saveFieldSchemaConfig() {
+    return element.all(by.css('.config.container')).all(by.cssContainingText('.btn', 'SAVE')).click();
+  }
+
+  saveGrokStatement() {
+    let saveButton = element(by.cssContainingText('metron-config-sensor-grok button', 'SAVE'));
+    return waitForElementVisibility(saveButton).then(() => {
+      return element(by.cssContainingText('metron-config-sensor-grok button', 'SAVE')).click();
+    })
+  }
+
+  saveThreatTriageRule() {
+    return element(by.cssContainingText('metron-config-sensor-rule-editor .btn', 'SAVE')).click();
+  }
+
+  setGrokStatement(statement: string) {
```

Comment:
This seems surprisingly sensor specific given the generality of the other calls. Are we presuming too much about the type of sensor here?